assets/src/stories-editor/helpers/index.js

```diff
@@ -1508,3 +1508,30 @@ export const getCallToActionBlock = ( pageClientId ) => {
 export const getUniqueId = () => {
 	return uuid().replace( /^\d/, 'a' );
 };
+
+/**
+ * Returns an image of the first frame of a given video.
+ *
+ * @todo Perhaps allow specifying wanted image type.
+ *
+ * @param {string} src Video src URL.
+ * @return {Promise<string>} The extracted image in base64-encoded format.
+ */
+export const getFirstFrameOfVideo = async ( src ) => {
+	const video = document.createElement( 'video' );
+	video.setAttribute( 'muted', 'muted' );
+
+	return new Promise( ( resolve ) => {
+		video.src = src;
+		video.addEventListener( 'loadeddata', () => {
+			const canvas = document.createElement( 'canvas' );
+			canvas.width = video.videoWidth;
+			canvas.height = video.videoHeight;
```

Comment:
Should we also consider the recommended media limits?

Reply:
I guess we can upload the original one and then let WordPress crop the image. I just adjusted `uploadVideoFrame` in the component to check for `MAX_IMAGE_SIZE_SLUG`.